scripts/make-cluster-account-secret.sh

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+
+print_usage() {
+  echo "Makes a kubernetes secret containing keys for a dscp account. The final line of this script will output a JSON object containing the new account's accountId."
+  echo ""
+  echo "Usage:"
+  echo "  ./scripts/make-cluster-account-secret.sh [ -h ] [ -f ] [ -n <namespace> ] [ -c <container> ] <cluster_name> <account_name>"
+  echo ""
+  echo "Example Usage:"
+  echo "  ./scripts/make-cluster-account-secret.sh inteli-stage api"
+  echo ""
+  echo "Flags: "
+  echo "  -h              Print this message"
+  echo "  -f              Force re-creation of new secret for an existing account. Note this is destructive"
+  echo "  -c <container>  Container image to use for key generation"
+  echo "  -n <namespace>  Namespace in which to create the secret. Defaults to dscp"
+}
+
+FORCE_RECREATE=
+NAMESPACE="dscp"
+CONTAINER="ghcr.io/digicatapult/vitalam-node:latest"
+while getopts ":n:c:fh" opt; do
+  case ${opt} in
+    h )
+      print_usage
+      exit 0
+      ;;
+    f )
+      FORCE_RECREATE="yes"
+      ;;
+    n )
+      NAMESPACE=${OPTARG}
+      ;;
+    c )
+      CONTAINER=${OPTARG}
+      ;;
+    : )
+      echo "Error: -${OPTARG} requires an argument." >&2
+      exit 1
+      ;;
+   \? )
+     echo "Invalid Option: -$OPTARG" 1>&2
+     echo "\n"
+     print_usage
+     exit 1
+     ;;
+  esac
+done
+shift $((OPTIND -1))
+CLUSTER=$1
+ACCOUNT_NAME=$2
+
+assert_cluster() {
+  local cluster=$1
+
+  printf "Checking cluster $cluster is configured correctly..." >&2
+  if [ ! -d "./clusters/$cluster/secrets" ]; then
+    echo -e "Cannot add secrets for cluster $cluster which does not exist" >&2
+    exit 1
+  fi
+  printf "OK\n" >&2
+}
+
+assert_not_account() {
+  local cluster=$1
+  local namespace=$2
+  local account_name=$3
+
+  printf "Checking account $account_name:$namespace does not already exist in cluster $cluster..." >&2
+  if [ ! -z "$FORCE_RECREATE" ]; then
+    printf "SKIP\n" >&2
+  else
+    if [ -f "./clusters/$cluster/secrets/${account_name}_${namespace}_account-keys.yaml" ] ||
+       [ -f "./clusters/$cluster/secrets/${account_name}_${namespace}_account-keys.unc.yaml" ]; then
+      echo -e "Account $account_name:${namespace} already exists in cluster $cluster. Use -f to overrite anyway." >&2
+      exit 1
+    fi
+    printf "OK\n" >&2
```

Comment:
just a question, why redirecting all stdout to stderror? should it be the other way around? 2>&1?

Reply:
means I can redirect stdout on the script for the final ouput